geonode/upload/forms.py

```diff
@@ -86,6 +95,24 @@ def clean(self):
         cleaned["valid_extensions"] = valid_extensions
         return cleaned
 
+    def validate_files_sum_of_sizes(self):
+        max_size = self._get_uploads_max_size()
+        total_size = sum([django_file.size for django_file in self.files.values()])
```

Comment:
@italogsfernandes this could raise some issues with zip files.
I was trying to upload a dataset, but something is not as expected, the Multidict value has X keyword as are the input files.
For example, the .zip file will have these two keywords:
```
'zip_file':<SimpleUploadedFile: ProvCM01012019_g.zip (application/zip)>
'base_file':<SimpleUploadedFile: ProvCM01012019_g.zip (application/zip)>
```
so with the some, the total size (assuming that the file is 1 byte) will be 2 bytes, while the file is only 1 byte.
My suggestion here could be ignore the "base_file" in the size sum in order to have a real and valid sum

Reply:
@italogsfernandes what about this?

Reply:
To do:
* [ ] Apply validation to `documents` upload page and add tests.
* [ ] Ignore `zip_file`, (`base_file` is used by other layer formats).

Reply:
@italogsfernandes i tried to upload again a document, but no exception is raised for the limit size (I set 1 byte with an image of 133.375 bytes) From UI it doesn't raise any exception, but it upload an empty resource and log this warn in console:

`Document has no field named 'name' - {'name': 'space.jpg', 'subtype': 'image', 'doc_url': None, 'extension': 'jpg', 'abstract': 'No abstract provided', 'date': datetime.datetime(2022, 1, 13, 9, 1, 7, 259632, tzinfo=<UTC>), 'date_type': 'Creation', 'bbox_polygon': None}`

can you give another check on it, please?

Reply:
It seems that your Upload size limit with slug=`document_upload_size` was set to the default value (100 MB) and your `file_upload_handler` size limit set to some small value.

The in the description of the `file_upload_handler` it says that the value should be greater than the other limits, but it may not be enough to prevent this kind of misconfiguration.

I gonna add a couple of validation methods to prevent this.